discovery-infra/test_infra/utils/utils.py

```diff
@@ -171,8 +171,8 @@ def set_tfvars(tf_folder, tfvars_json):
         json.dump(tfvars_json, _file)
 
 
-def are_hosts_in_status(hosts, nodes_count, statuses, fall_on_error_status=True):
-    hosts_in_status = [host for host in hosts if host["status"] in statuses]
+def are_hosts_in_status(hosts, nodes_count, statuses, status_info="", fall_on_error_status=True):
+    hosts_in_status = [host for host in hosts if (host["status"] in statuses and status_info in host["status_info"])]
```

Comment:
This should work.
Consider
```suggestion
    hosts_in_status = [host for host in hosts if (host["status"] in statuses and host["status_info"].startswith(status_info))]
```

Reply:
this one works well when running test with SNO, however its fail when running multi-node (both tests using same method and strings). I need to deep dive it to understand why, but before doing so, is there a big difference between `startswith` and `status_info in` ? how critical is it?

Reply:
done